action/protocol/staking/handlers.go

```diff
@@ -597,30 +627,46 @@ func (p *Protocol) handleCandidateRegister(ctx context.Context, act *action.Cand
 	}
 
 	if err := csm.Upsert(c); err != nil {
-		return log, csmErrorToHandleError(owner.String(), err)
+		return log, nil, csmErrorToHandleError(owner.String(), err)
 	}
 
 	// update caller balance
 	if err := caller.SubBalance(act.Amount()); err != nil {
-		return log, &handleError{
+		return log, nil, &handleError{
 			err:           errors.Wrapf(err, "failed to update the balance of register %s", actCtx.Caller.String()),
 			failureStatus: iotextypes.ReceiptStatus_ErrNotEnoughBalance,
 		}
 	}
 	// put updated caller's account state to trie
 	if err := accountutil.StoreAccount(csm, actCtx.Caller, caller); err != nil {
-		return log, errors.Wrapf(err, "failed to store account %s", actCtx.Caller.String())
+		return log, nil, errors.Wrapf(err, "failed to store account %s", actCtx.Caller.String())
 	}
 
 	// put registrationFee to reward pool
 	if err := p.depositGas(ctx, csm, registrationFee); err != nil {
-		return log, errors.Wrap(err, "failed to deposit gas")
+		return log, nil, errors.Wrap(err, "failed to deposit gas")
 	}
 
 	log.AddAddress(owner)
 	log.AddAddress(actCtx.Caller)
 	log.SetData(byteutil.Uint64ToBytesBigEndian(bucketIdx))
-	return log, nil
+
+	// generate candidate register log
+	rewardingAddr := hash.Hash160b([]byte(action.RewardingProtocolID))
+	cLog := action.Log{
+		Address: p.addr.String(),
+		Topics: action.Topics{
+			action.CandidateRegistrationFee,
+			hash.BytesToHash256(actCtx.Caller.Bytes()),
+			hash.BytesToHash256(rewardingAddr[:]),
```

Comment:
from action caller to rewarding protocol address

Reply:
also need one from action caller to staking protocol for bucket creation